controller/src/controller/db/sql/sqlclient.py

```diff
@@ -30,10 +30,10 @@ class SqlClient(Client):
     This is the SQL client that interact with the SQL database.
     """
 
-    def __init__(self, db_url: str, verbose: bool = False):
-        self.db_url = db_url
+    def __init__(self, url: str, verbose: bool = False):
```

Comment:
notice you give `"db_url"` in the config, and you pass `**config.db` to the init, so renaming to `url` will break it.